src/Compilers/CSharp/Portable/Compilation/MemberSemanticModel.cs

```diff
@@ -1087,6 +1087,17 @@ public override TypeInfo GetTypeInfo(SelectOrGroupClauseSyntax node, Cancellatio
         private void GetBoundNodes(CSharpSyntaxNode node, out CSharpSyntaxNode bindableNode, out BoundNode lowestBoundNode, out BoundNode highestBoundNode, out BoundNode boundParent)
         {
             bindableNode = this.GetBindableSyntaxNode(node);
+
```

Comment:
@cston I dont think GetBindableSyntaxNode strictly only walk up spine (http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp/Compilation/MemberSemanticModel.cs,1720) it sometimes, move to its children which can make it moves to different spine.

if you sync this PR and remove that check and run TestParentOperations from Roslyn.Compilers.CSharp.Semantic.UnitTests, you can live repro the issue.

I also have put dump in ([mlang share]\public\hechang\cton) for quick look.